libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/drillDownUtil.ts

```diff
@@ -0,0 +1,165 @@
+            }
+
+            return [...acc, cur];
+        },
+    );
+
+    return insightModifyItems(
+        removedLeftAttributes,
+        (bucketItem: IAttributeOrMeasure): IAttributeOrMeasure => {
+            if (isAttribute(bucketItem) && matchesDrillDownTargetAttribute(drillDefinition, bucketItem)) {
+                const displayForm = drillDownDisplayForm(drillDefinition);
+                return modifyAttribute(bucketItem, (a) => a.displayForm(displayForm).noAlias());
+            }
+            return bucketItem;
+        },
+    );
+}
+
+function removePropertiesForRemovedAttributes(insight: IInsight): IInsight {
+    const properties: VisualizationProperties = insightProperties(insight);
+
+    if (!properties) {
+        return insight;
+    }
+
+    const identifiers = insightItems(insight).map((bucketItem: IAttributeOrMeasure) =>
+        bucketItemLocalId(bucketItem),
+    );
+
+    const result = Object.entries(properties).reduce((acc, [key, value]) => {
+        if (key === ENUM_PROPERTIES_TYPE.CONTROLS && value.columnWidths) {
+            const columns = value.columnWidths.filter(
+                (c: TColumnWidths) =>
+                    c?.attributeColumnWidthItem?.attributeIdentifier === undefined ||
+                    identifiers.includes(c.attributeColumnWidthItem.attributeIdentifier),
+            );
+
+            return {
+                ...acc,
+                [key]: {
+                    columnWidths: columns,
+                },
+            };
+        }
+
+        return { ...acc };
+    }, properties);
+
+    return insightSetProperties(insight, result);
+}
+
+export function sanitizeTableProperties(insight: IInsight): IInsight {
+    return removePropertiesForRemovedAttributes(insight);
+}
+
+export function convertIntersectionToFilters(intersections: IDrillEventIntersectionElement[]): IFilter[] {
+    return intersections
+        .map((intersection) => intersection.header)
+        .filter(isDrillIntersectionAttributeItem)
+        .map((header) =>
+            newPositiveAttributeFilter(uriRef(header.attributeHeader.uri), {
+                uris: [header.attributeHeaderItem.uri],
+            }),
+        );
+}
+
+export function addIntersectionFiltersToInsight(
+    source: IInsight,
+    intersection: IDrillEventIntersectionElement[],
+): IInsight {
+    const filters = convertIntersectionToFilters(intersection);
+    const resultFilters = [...source.insight.filters, ...filters];
+
+    return insightSetFilters(source, resultFilters);
+}
+
+export function getIntersectionPartAfter(
+    intersection: IDrillEventIntersectionElement[],
+    localIdentifier: string,
+): IDrillEventIntersectionElement[] {
+    const index = intersection.findIndex(
+        (item: IDrillEventIntersectionElement) =>
+            isDrillIntersectionAttributeItem(item.header) &&
+            item.header.attributeHeader.localIdentifier === localIdentifier,
+    );
+
+    return intersection.slice(index);
+}
```

Comment:
seems like this should be closer to the drilling code that we have in `sdk-ui`

Reply:
Moved